client/pages/post-form.jsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import SideSelectButton from '../components/side-select-button';
+
+export default class PostForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      side: 'White',
+      message: ''
+    };
+    this.handleSelect = this.handleSelect.bind(this);
+    this.handleMessage = this.handleMessage.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSelect(event) {
+    if (event.target.classList.contains('white')) {
+      this.setState({ side: 'White' });
+    } else if (event.target.classList.contains('black')) {
+      this.setState({ side: 'Black' });
+    }
+  }
+
+  handleMessage(event) {
+    this.setState({ message: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const button = event.nativeEvent.submitter;
+    if (button.classList.contains('side-select')) {
+      return;
+    }
+    if (button.classList.contains('cancel-btn')) {
+      window.location.hash = '#join';
+      return;
+    }
+
+    const body = {
+      playerName: 'Anonymous',
+      message: this.state.message,
+      playerSide: this.state.side.toLowerCase()
+    };
+    const req = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(body)
+    };
+    fetch('/api/create-post', req)
+      .then(res => res.json())
+      .then(result => {
+        this.props.onCreate(result);
+        window.location.hash = '#game';
```

Comment:
I strongly recommend including the game's id in the hash, so that users can join a specific game that is recorded in the browser history.
window.location.hash = `#game?gameId=${result.gameIdOrWhatever}`;